pages/golfers/[id].js

```diff
@@ -0,0 +1,50 @@
+import useUserScores from '../../lib/useUserScores'
+import Layout from '../../components/Layout'
+import ScoreCard from '../../components/ScoreCard'
+import useUserName from '../../lib/useUserName'
+import { useRouter } from 'next/router'
+import { useEffect, useState } from 'react'
+
+
+const UserScores = () => {
+  const router = useRouter()
+  const { id } = router.query
+  const { name, error: errorUser } = useUserName(id)
+  const { scores, error: errorScores } = useUserScores(id)
```

Comment:
Go to this page, open browser dev tools (inspect element), navigate to the Network tab. Is there any `undefined` requests? (you might need to refresh the page having the network tab opened)

Reply:
Yes, why is this happening?

Reply:
Because the initial value that `router.query` returns for the `id` is undefined. We should treat this case at some level so that we're not triggering a request if the `id` is `undefined`. Be careful as hooks cannot be conditionally declared.